tests/failpoints_cases/test_snap.rs

```diff
@@ -189,3 +189,59 @@ fn test_server_snapshot_on_resolve_failure() {
     // Clean up.
     fail::remove(on_resolve_fp);
 }
+
+#[test]
+fn test_generate_snapshot() {
+    let _guard = ::setup();
+
+    let mut cluster = new_server_cluster(1, 7);
```

Comment:
I think use 3 nodes and 2 failpoints is enough. run_conf_change -> put k1 -> pause delete -> add node 2 -> check k1 -> pause do build -> add node 3 -> check generating -> resume delete -> resume do build -> check k1 exists on node 3.